.github/workflows/tests.yml

```diff
@@ -486,6 +486,194 @@ jobs:
           name: lxd-clients-${{ runner.os }}
           path: bin/
 
+  ui:
+    name: UI e2e tests
+    runs-on: ubuntu-latest
+    needs: [code-tests, documentation]
+    if: ${{ github.repository == 'canonical/lxd' && github.event_name == 'push' }}
+    env:
+      LXD_DIR: "/var/lib/lxd"
+      LXD_OIDC_CLIENT_ID: ${{ secrets.LXD_UI_OIDC_CLIENT_ID }}
+      LXD_OIDC_ISSUER: ${{ secrets.LXD_UI_OIDC_ISSUER }}
+      LXD_OIDC_AUDIENCE: ${{ secrets.LXD_UI_OIDC_AUDIENCE }}
+      LXD_OIDC_USER: ${{ secrets.LXD_UI_OIDC_TEST_USER }}
+      LXD_OIDC_PASSWORD: ${{ secrets.LXD_UI_OIDC_TEST_PASSWORD }}
+      LXD_OIDC_GROUPS_CLAIM: "lxd-idp-groups"
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Tune disk performance
+        uses: ./.github/actions/tune-disk-performance
+
+      - name: Install Go
+        uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed # v5.1.0
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Install dependencies for running lxd
```

Comment:
I've just reworked the actions so you can now replace that bit with:
```
      - name: Install runtime dependencies
        uses: ./.github/actions/install-lxd-runtimedeps
```

Reply:
I am adding some old dependencies for dqlite back to encounter this issue:

![image](https://github.com/user-attachments/assets/27d664cf-3312-42a4-995c-15d01b51fa45)

Reply:
That `libdqlite.so.0` should be coming from the system-test-deps artifact, no?

Reply:
All I know is with [this changeset](https://github.com/canonical/lxd/compare/58ba9f6f9d15574dc6d2f683a141711504b03e4d..5fdd983438347b63cb4077080ba4c64548a8e8a5) the UI e2e test suite completed. Before without those extra deps I get the error from the screenshot above.

Reply:
I think this is all tied to a wrong `LD_LIBRARY_PATH` and your change works by accident as it pulls in `libdqlite-dev` which happens to depend on `libdqlite0` that has a (random) copy of that `libdqlite.so.0`.

Fixing the `LD_LIBRARY_PATH` should make any modification of the actions unneeded. I realize there is a lot of back-n-forth and that must be frustrating for you so I'd like to propose doing a Meet tomorrow if you think that can help/speed things up.

Reply:
You are absolutely right, thx for the tip! I can see it worked with the fixed path in [this fork run](https://github.com/edlerd/lxd/actions/runs/11967855652/job/33366069682).